llvm/test/Transforms/LoopVectorize/X86/drop-poison-generating-flags.ll

```diff
@@ -563,29 +563,24 @@ define void @pr70590_recipe_without_underlying_instr(i64 %n, ptr noalias %dst) {
 ; CHECK-NEXT:    [[TMP3:%.*]] = extractelement <4 x i1> [[TMP2]], i32 0
 ; CHECK-NEXT:    br i1 [[TMP3]], label [[PRED_SREM_IF:%.*]], label [[PRED_SREM_CONTINUE:%.*]]
 ; CHECK:       pred.srem.if:
-; CHECK-NEXT:    [[TMP4:%.*]] = srem i64 3, 0
 ; CHECK-NEXT:    br label [[PRED_SREM_CONTINUE]]
 ; CHECK:       pred.srem.continue:
-; CHECK-NEXT:    [[TMP5:%.*]] = phi i64 [ poison, %vector.body ], [ [[TMP4]], [[PRED_SREM_IF]] ]
```

Comment:
Can we have a variant of the test where we don't just use the first lane of the simplified replicate recipe, e.g. storing the result outside the replicate region?

Would also be good to have a case where we don't fold to poison, but I am not sure if that's possible.

In that case, could we constant-fold `VPPredPhi poison -> poison`?

Reply:
I've added the additional test, but constant-folding PredPHI seems to be out of scope since it doesn't have an opcode? Could address in a follow-up.

Reply:
Yep, that would need to extra code to handle `PredPhi poison -> poison`. If that would be enough to remove the need to change `VPPredInstPHIRecipe::execute(` and. `VPTransformState::` that would be desireable.

Reply:
That's a good idea, thanks! I've handled PredPHI constant -> constant, and cleaned up the changes in VPlan(Recipes).